contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+    uint256 public constant BASIS_POINTS_GRANULARITY = 10_000;
+
+    /// @notice The StableSwap pool to deposit in
+    address public pool;
+
+    // ------------------ Private properties -----------------------------------
+
+    /// some fixed variables to interact with the pool
+    uint8 private _feiIndex; // index of FEI in the pool (0 or 1)
+    uint8 private _3crvIndex; // index of 3crv in the pool (0 or 1)
+    address private _3crv; // address of the 3crv token
+    address private _3pool; // address of the 3pool
+    address private _dai; // address of the DAI token
+    address private _usdc; // address of the USDC token
+    address private _usdt; // address of the USDT token
+
+    // ------------------ Constructor ------------------------------------------
+
+    /// @notice Curve PCV Deposit constructor
+    /// @param _core Fei Core for reference
+    /// @param _pool StableSwap to deposit to
+    /// @param _curve3pool the Curve 3pool
+    /// @param _depositMaxSlippageBasisPoints max slippage for deposits, in bp
+    constructor(
+        address _core,
+        address _pool,
+        address _curve3pool,
+        uint256 _depositMaxSlippageBasisPoints
+    ) CoreRef(_core) {
+        // public variables
+        pool = _pool;
+        depositMaxSlippageBasisPoints = _depositMaxSlippageBasisPoints;
+
+        // cached private variables
+        _3crvIndex = IStableSwap(pool).coins(0) == address(fei()) ? 1 : 0;
+        _feiIndex = _3crvIndex == 0 ? 1 : 0;
+        _3crv = IStableSwap(pool).coins(_3crvIndex);
+        _3pool = _curve3pool;
+        _dai = IStableSwap(_3pool).coins(0);
+        _usdc = IStableSwap(_3pool).coins(1);
+        _usdt = IStableSwap(_3pool).coins(2);
+    }
+
+    /// @notice deposit DAI, USDC, USDT, 3crv, and FEI into the pool.
+    /// Note: the FEI has to be minted & deposited on this contract in a previous
+    ///       tx, as this contract does not use the Minter role.
+    function deposit() public override onlyPCVController whenNotPaused {
+        IFei _fei = fei();
+        uint256 _3crvVirtualPrice = IStableSwap(_3pool).get_virtual_price();
+
+        // Deposit DAI, USDC, and USDT, to get 3crv
+        uint256 _daiBalance = IERC20(_dai).balanceOf(address(this));
+        uint256 _usdcBalance = IERC20(_usdc).balanceOf(address(this));
+        uint256 _usdtBalance = IERC20(_usdt).balanceOf(address(this));
+        uint256 _3crvBalanceBefore = IERC20(_3crv).balanceOf(address(this));
+        if (_daiBalance != 0 || _usdcBalance != 0 || _usdtBalance != 0) {
+            uint256[3] memory _add3poolLiquidityAmounts;
+            _add3poolLiquidityAmounts[0] = _daiBalance;
+            _add3poolLiquidityAmounts[1] = _usdcBalance;
+            _add3poolLiquidityAmounts[2] = _usdtBalance;
+            IERC20(_dai).approve(_3pool, _daiBalance);
+            IERC20(_usdc).approve(_3pool, _usdcBalance);
+            SafeERC20.safeApprove(IERC20(_usdt), _3pool, _usdtBalance);
```

Comment:
Why not use safeApprove for all 3?

Reply:
Only USDT required it because it doesn't have a standard return, but I'll use it for all 3, for uniformity